stdlib/LinearAlgebra/src/structuredbroadcast.jl

```diff
@@ -64,7 +64,39 @@ find_bidiagonal() = throw(ArgumentError("could not find Bidiagonal within broadc
 find_bidiagonal(a::Bidiagonal, rest...) = a
 find_bidiagonal(bc::Broadcast.Broadcasted, rest...) = find_bidiagonal(find_bidiagonal(bc.args...), rest...)
 find_bidiagonal(x, rest...) = find_bidiagonal(rest...)
+
+find_uplo(a::Bidiagonal) = a.uplo
+find_uplo(a::Char) = a
+find_uplo(a::Number) = Nothing
+find_uplo(a) = Nothing
+
+function find_uplo(bc::Broadcast.Broadcasted, rest...)
+
+    left = find_uplo(bc.args[1])
+    right = find_uplo(bc.args[2])
+
+    tmp = (left, right)
```

Comment:
This could be zero, one, two, or more arguments here.  Here's what I think I'd do:

```Julia
merge_uplos(::Nothing, ::Nothing) = nothing
merge_uplos(::Nothing, b) = b
merge_uplos(a, ::Nothing) = a
merge_uplos(a, b) = a == b ? a : 'T'

find_uplo(a) = nothing
find_uplo(a::Bidiagonal) = a.uplo
find_uplo(bc::Broadcasted) = mapreduce(find_uplo, merge_uplos, bc.args, init=nothing)
```

Reply:
This is awesome!! So clean. Changed to this in aec7bee